tools/chplcheck/src/rules.py

```diff
@@ -323,3 +323,111 @@ def variables(node):
+            return isinstance(node, (Record, Class, Module, Block))
+
+        def unwrap_intermediate_block(node):
+            """
+            Given a node, find the reference indentation that its children
+            should be compared against.
+
+            This method also rules out certain nodes that should
+            not be used for parent-child indentation comparison.
+            """
+            if not isinstance(node, Block):
+                return node
+
+            parent = node.parent()
+            if not parent:
+                return node
+
+            if isinstance(parent, (Function, Loop)):
+                return parent
+            elif isinstance(parent, Conditional) and parent.is_expression_level():
+                return None
+            return node
+
+        # If root is something else (e.g., function call), do not
+        # apply indentation rules; only apply them to things that contain
+        # a list of statements.
+        is_eligible_parent_for_indentation = contains_statements(root)
+        if not is_eligible_parent_for_indentation:
+            return
+
+        parent_for_indentation = unwrap_intermediate_block(root)
+        parent_depth = None
+        if parent_for_indentation is None:
+            # don't compare against any parent depth.
+            pass
+
+        # For implicit modules, proper code will technically be on the same
+        # line as the module's body. But we don't want to warn about that,
+        # since we don't want to ask all code to be indented one level deeper.
+        elif not (isinstance(parent_for_indentation, Module) and parent_for_indentation.kind() == "implicit"):
+            parent_depth = parent_for_indentation.location().start()[1]
+
+        prev = None
+        prev_depth = None
+        prev_line = None
+
+        # We only care about misaligned statements, so we don't want to do stuff
+        # like warn for inherit-exprs or pragmas on a record.
+        iterable = root
+        if isinstance(root, AggregateDecl):
+            iterable = root.decls_or_comments()
+
+        for child in iterable:
+            if isinstance(child, Comment): continue
+
+            # NamedDecl nodes currently use the name as the location, which
+            # does not indicate their actual indentation.
+            if isinstance(child, (NamedDecl, TupleDecl, ForwardingDecl)): continue
+
+            (line, depth) = child.location().start()
+
+            # Warn for two statements on one line:
+            #   var x: int; var y: int;
+            if line == prev_line:
+                yield child
+
+            # Warn for misaligned siblings:
+            #   var x: int;
+            #     var y: int;
+            elif prev_depth and depth != prev_depth:
+                # Special case, slightly coarse: avoid double-warning with
+                # MisleadingIndentation
+                if not(prev and isinstance(prev, Loop) and prev.block_style() == "implicit"):
```

Comment:
This indicates to me that maybe these two rules could be merged in the future, but I would consider that future work

Reply:
I thought about this, but I would actually prefer them to report different names. Which means that there might be some architectural considerations / changes to support that if we did want to fuse them.

Reply:
What is it about MisleadingIndentation that makes it special? To my mind, we now have two rules that do very similar things. Whats the advantage of them being separate?

Reply:
I think it's more specific, and indicates a misunderstanding of Chapel's non-indentation-sensitive nature. Meanwhile, the less specific warning is just "your editor has tabwidth=4 instead of 8" (say).